src/Messaging/Handlers/PeerMessageHandler.cs

```diff
@@ -324,16 +324,12 @@ public async void HandleMessageRead(object sender, byte[] message)
 
                     case MessageCode.Peer.UploadFailed:
                         var uploadFailedResponse = UploadFailed.FromByteArray(message);
-                        var msg = $"Download of {uploadFailedResponse.Filename} reported as failed by {connection.Username}";
-
-                        var download = SoulseekClient.DownloadDictionary.Values.FirstOrDefault(d => d.Username == connection.Username && d.Filename == uploadFailedResponse.Filename);
-                        if (download != null)
```

Comment:
this check isn't necessary; nothing will happen if the thrown wait doesn't exist